### PR TITLE
Use spf13/cobra for docker login and docker logout

### PR DESCRIPTION
**Pull request by vdemeester**
Moves _registry_ command login and logout to `api/client/registry/login.go` and `api/client/registry/logout.go` and use cobra  🐍.

Also moved some _utils_ method in better files.

/cc @dnephin @thaJeztah @LK4D4 @cpuguy83 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: cli/required.go [hunk 1]
```diff
@@ -43,6 +43,23 @@ func RequiresMinArgs(min int) cobra.PositionalArgs {
 	}
 }
 
+// RequiresMaxArgs returns an error if there is not at most min args
+func RequiresMaxArgs(min int) cobra.PositionalArgs {
```
Comment (dnephin) [inline cli/required.go]: The parameter should probably be `max` otherwise it's kind of confusing

Comment (vdemeester) [inline cli/required.go]: oh damn you're absolutely right :sweat: 
/me was in *robot* mode I guess :sweat_smile:

File: api/client/login.go [hunk 2]
```diff
@@ -1,184 +0,0 @@
...
-
-	// Deprecated in 1.11: Should be removed in docker 1.13
-	cmd.String([]string{"#e", "#-email"}, "", "Email")
-
-	cmd.ParseFlags(args, true)
-
-	// On Windows, force the use of the regular OS stdin stream. Fixes #14336/#14210
-	if runtime.GOOS == "windows" {
-		cli.in = os.Stdin
-	}
```
Comment (dnephin) [inline api/client/login.go]: Is this not necessary anymore?

Comment (vdemeester) [inline api/client/login.go]: ah no it is still necessary :stuck_out_tongue_closed_eyes:

Comment (vdemeester) [inline api/client/login.go]: Given where it was used, moved it there : https://github.com/docker/docker/pull/23265/files#diff-2b2e46176d739e8721f07f3d736e13f8R86

**Comment by mlaventure** [comment]
This already needs a rebase unfortunately @vdemeester

**Comment by dmcgowan** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM

ping @runcom PTAL as well

**Comment by thaJeztah** [comment]
also, I think the commits should be squashed, because some changes are in the wrong commit

**Comment by runcom** [comment]
LGTM but this needs a rebase :(

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**